platform/MCU/STM32F2xx/STM32_USB_Device_Driver/src/usbd_flash_if.c

```diff
@@ -155,6 +160,15 @@ uint16_t FLASH_If_Write(uint32_t Add, uint32_t Len)
   /* Lock the internal flash */
   FLASH_Lock();
 
+  /* This make sure that after the system parts being updated,
+   * the address of DCT functions that resided in system parts are up to date,
+   * otherwise, bootloader cannot access the DCT after updating the system parts without a reset. */
+#if PLATFORM_ID == PLATFORM_PHOTON_PRODUCTION || PLATFORM_ID == PLATFORM_P1
+#if MODULE_FUNCTION == MOD_FUNC_BOOTLOADER
+  dct_reload_functions();
```

Comment:
Let's also add a check on the address and length and trigger the call to this function only when modifying system-part1 and system-part2 area.

Reply:
I'm afraid that we can only check the address, but not the length. The length being passed into the `FLASH_If_Write()` function is for the current small data block.

Reply:
:+1: That's fine.